src/node/node_info_network.h

```diff
@@ -37,12 +46,15 @@ namespace ccf
       std::optional<size_t> max_open_sessions_soft = std::nullopt;
       std::optional<size_t> max_open_sessions_hard = std::nullopt;
 
+      EndorsementType endorsement_type = EndorsementType::Network;
```

Comment:
As this is a new field in a KV table (that automatically ends up in the ledger), could you declare the field as `std::optional` so that it is clear when accessing it that it may not be present (i.e. for historical ledgers). We've followed this approach so far, and should document it under https://microsoft.github.io/CCF/main/contribute/index.html.